.github/workflows/appsignals-e2e-k8s-test.yml

```diff
@@ -0,0 +1,220 @@
+## Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+## SPDX-License-Identifier: Apache-2.0
+
+# This is a reusable workflow for running the E2E test for App Signals.
+# It is meant to be called from another workflow.
+# Read more about reusable workflows: https://docs.github.com/en/actions/using-workflows/reusing-workflows#overview
+name: App Signals Enablement E2E Testing - K8s on EC2 Use Case
+on:
+  workflow_call:
+    inputs:
+      aws-region:
+        required: true
+        type: string
+      caller-workflow-name:
+        required: true
+        type: string
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  # The presence of this env var is required for use by terraform and AWS CLI commands
+  # It is not redundant
+  AWS_DEFAULT_REGION: ${{ inputs.aws-region }}
+  TEST_ACCOUNT: ${{ secrets.APP_SIGNALS_E2E_TEST_ACC }}
+  METRIC_NAMESPACE: AppSignals
+  LOG_GROUP_NAME: /aws/appsignals/k8s
+  TEST_RESOURCES_FOLDER: /home/runner/work/aws-application-signals-test-framework/aws-application-signals-test-framework
+  MASTER_NODE_SSH_KEY: ${{ secrets.APP_SIGNALS_E2E_K8S_SSH_KEY_IAD }}
+  MAIN_SERVICE_ENDPOINT: ${{ secrets.APP_SIGNALS_E2E_K8S_MASTER_NODE_ENDPOINT }}
```

Comment:
Why does the `MAIN_SERVICE_ENDPOINT` need to be a secret?
What's the plan if we want run the canary on extra regions?

Reply:
After looking at the Terraform code makes sense. There is a pre-existing ec2 instance and k8s we use to run the test. I'm assuming it's difficult to setup the ec2 instance and K8s each run?

Reply:
Exactly. Since we are reusing a single EC2 instance we wouldn't want someone to send traffic to it randomly or have access to the public endpoint for no reason. It's not that important since it's public but just helps as an extra measure to avoid it being easily knowable.

Also, this implementation will help us in the future if we want to extend it to cover more regions since we can pass in the values through, for example, a matrix input.